services/QuillLMS/app/controllers/teachers/progress_reports/activity_sessions_controller.rb

```diff
@@ -13,14 +13,31 @@ def index
           flash[:warning] = 'Downloadable reports are only available to Premium users.'
           return redirect_to premium_path
         end
-        return_data(false)
+        render plain: fetch_index_cache(false)
       end
       format.json do
-        return_data(true)
+        render json: fetch_index_cache(true)
       end
     end
   end
 
+  private def fetch_index_cache(as_json)
```

Comment:
I feel like `as_json` is kind of a confusing name for a boolean since there's `obj.as_json` which returns a hash.  Maybe `is_json`?